packages/eslint-plugin/tests/rules/no-floating-promises.test.ts

```diff
@@ -475,6 +475,43 @@ Promise.reject()
   .finally(() => {});
       `,
     },
+    {
+      code: `
+await Promise.all([Promise.resolve(), Promise.resolve()]);
+      `,
+    },
+    {
+      code: `
+declare const promiseArray: Array<Promise<unknown>>;
+void promiseArray;
+      `,
+    },
+    {
+      // This one is a bit of a head-scratcher, knowing that the rule recursively
+      // checks .finally() expressions. However, ultimately, the whole expression
+      // is not a promise (it's invalid TS), so it's not a floating promise.
+      code: `
```

Comment:
[Testing] Fun 😄 but I don't think we need to include test cases for invalid type errors. Since Array doesn't have a `.finally`, I don't think there's a need to test for the user trying it.